components/calendar/calendar.js

```diff
@@ -0,0 +1,474 @@
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.addEventListener('keydown', this._onKeyDown);
+		this.addEventListener('d2l-calendar-selected', (e) => {
+			this._selectedDate = e.detail.date;
+		});
+
+		let selected;
+		if (this.selectedValue) {
+			selected = parseDate(this.selectedValue);
+		} else {
+			selected = new Date();
+		}
+
+		this._selectedDate = {
+			month: selected.getMonth(),
+			date: selected.getDate(),
+			year: selected.getFullYear()
+		};
+
+		this._shownMonth = this._selectedDate.month;
+		this._shownYear = this._selectedDate.year;
+
+		let focusDay = 1;
+		if (this._selectedDate.month === this._shownMonth) focusDay = this._selectedDate.date;
+		this._focusDate = {
+			date: focusDay,
+			month: this._shownMonth,
+			year: this._shownYear
+		};
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === '_selectedDate') {
+				this._generateDaysInMonth();
+			}
+			if (prop === '_shownMonth') {
+				this._onMonthChange();
+				this._generateDaysInMonth();
+				this.updateComplete.then(() => {
+					this.updateComplete.then(() => {
+						this._getNodeAndAddFocus(this._keyboardTriggeredMonthChange);
+						this._keyboardTriggeredMonthChange = false;
+					});
+				});
+			}
+		});
+	}
+
+	render() {
+		const weekdayHeaders = this._arrangedWeekdays.map((day) => html`
+			<th role="columnheader" abbr="${day.long}">
+				<abbr title="${day.long}">
+					${day.short}
+				</abbr>
+			</th>
+		`);
+
+		let dayRows;
+		if (this._dates) {
+			dayRows = this._dates.map((week) => {
+				const weekHtml = week.map((day) => {
+					return html`
+						<td>
+							<d2l-calendar-date
+								date=${day.date}
+								month=${day.month}
+								year=${day.year}
+								?focused=${day.focused}
+								?other-month=${day.otherMonth}
+								?selected=${day.selected}>
+							</d2l-calendar-date>
+						</td>`;
+				});
+
+				return html`
+					<tr>${weekHtml}</tr>
+				`;
+			});
+		}
+		const heading = `${this._monthNames[this._shownMonth]} ${this._shownYear}`;
+		return html`
+			<div aria-labelledby="${this._dialogLabelId}" class="d2l-calendar">
+				<table summary="${this.summary}" role="grid">
+					<thead>
+						<tr class="d2l-calendar-title">
+							<td>
+								<d2l-button-icon @click="${this._showPrevMonth}" text="${this._computeText(this._prevMonth)}" icon="tier1:chevron-left"></d2l-button-icon>
+							</td>
+							<th colspan="5">
+								<h2 class="d2l-heading-4" aria-live="polite" id="${this._dialogLabelId}">${heading}</h2>
+							</th>
+							<td>
+								<d2l-button-icon @click="${this._showNextMonth}" text="${this._computeText(this._nextMonth)}" icon="tier1:chevron-right"></d2l-button-icon>
+							</td>
+						</tr>
+					</thead>
+					<tbody>
+						<tr>
+							${weekdayHeaders}
+						</tr>
+						${dayRows}
+					</tbody>
+				</table>
+				<slot></slot>
```

Comment:
Is the slot used for anything?

Reply:
I think it will be used in the future when we need to add the "Set to Today" and "Clear" buttons, but that could wait until a future PR as it's a separate story. Either way!